malaria-ios/NSManagedObjects/User.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+import CoreData
+
+class User: NSManagedObject {
+  
+  static private let MinimumCharacterLength = 2
+  static private let MinimumVolunteerAge = 12
+  static private let MaximumVolunteerAge = 90
+  
+  enum UserValidationError: String, ErrorType {
+    case NoFirstNameProvided = "Please insert your first name."
+    case NoLastNameProvided = "Please insert your last name."
+    case NoAgeProvided = "Please insert your age."
+    case NoEmailProvided = "Please insert your email."
+    
+    case FirstNameLength = "First name must have minimum two characters."
+    case LastNameLength = "Last name must have minimum two characters."
+    case LocationLength = "Location must have minimum two characters."
+    case PhoneLength = "Phone must have minimum two characters."
+    case GenderLength = "Gender must have minimum two characters."
+    
+    case InvalidAge = "Please insert real age."
+    case InvalidEmail = "Please insert a valid email."
+    case InvalidPhone = "Please insert a valid phone number."
+  }
+  
+  static func isUserAlreadyCreated() -> Bool {
+    let newContext = CoreDataHelper.sharedInstance.createBackgroundContext()
+    return User.retrieve(User.self, context: newContext!).count > 0
+  }
+  
+  /**
+   Defines or updates current user. It makes sure only one user exists at a time.
+   
+   - return MethodReturnType: A tuple which can contain the created user or,
+   if the user creation failed, the corespondent error as `String`.
+   */
+  
+  static func define(firstName: String?,
+                     lastName: String?,
+                     age: String?,
+                     email: String?,
+                     gender: String?,
+                     location: String?,
+                     phone: String?) throws -> User {
+    
+    do {
+      try validateData(firstName,
+                       lastName: lastName,
+                       age: age,
+                       location: location,
+                       email: email,
+                       gender: gender,
+                       phone: phone)
+    }
+    catch let error as UserValidationError {
```

Comment:
you don't need to catch **just** to rethrow. Remove the try catch and add `throws tag in the define method`